currint/currency.py

```diff
@@ -101,10 +101,17 @@ def format_decimal(self, value):
         Formats a value (in the minor unit as an integer) as a string,
         with no local prefix/suffix.  Can be cast into a Decimal value as needed.
         """
-        major_int = int(self.minor_to_major(value))
-        minor_int = abs(value) % self.divisor
-        format_str = "%i.%0" + str(self.exponent or 0) + "i"
-        return format_str % (major_int, minor_int)
+        major_value = self.minor_to_major(value)
+
+        if self.exponent is not None:
```

Comment:
maybe avoid the extra assignments and make is a little more simple
```
if self.exponent:
    return '{:.{precision}f}'.format(major_value, precision=self.exponent)

# A custom `divisor` can produce arbitrary precision major representations (think divisor=13)
return '{:f}'.format(major_value)
```

or named
```
if self.exponent:
    return '{major_value:.{precision}f}'.format(
        major_value=major_value, 
        precision=self.exponent,
    )

# A custom `divisor` can produce arbitrary precision major representations (think divisor=13)
return '{major_value:f}'.format(
    major_value=major_value, 
)
```